lib/puppet/provider/apt_key/apt_key.rb

```diff
@@ -41,14 +49,17 @@ def self.instances
       end
 
       new(
-        :name    => line_hash[:key_id],
-        :id      => line_hash[:key_id],
-        :ensure  => :present,
-        :expired => expired,
-        :expiry  => line_hash[:key_expiry],
-        :size    => line_hash[:key_size],
-        :type    => line_hash[:key_type] == 'R' ? :rsa : :dsa,
-        :created => line_hash[:key_created]
+        :name        => line_hash[:key_fingerprint],
+        :id          => line_hash[:key_fingerprint],
```

Comment:
True, but short is bad (not nearly unique enough).  So I figured it had to change to long or fingerprint and chose fingerprint.  It's a simple change back to short if it causes some breaking behavior (though I couldn't find anything that uses it).